browser/resources/settings/brave_search_engines_page/private_search_engine_list_dialog.ts

```diff
@@ -0,0 +1,112 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import 'chrome://resources/cr_elements/cr_button/cr_button.js'
+import 'chrome://resources/cr_elements/cr_radio_button/cr_radio_button.js'
+import 'chrome://resources/cr_elements/cr_radio_group/cr_radio_group.js'
+import 'chrome://resources/cr_elements/cr_shared_style.css.js'
+import 'chrome://resources/cr_elements/cr_shared_vars.css.js'
+import '../settings_shared.css.js'
+import '../site_favicon.js'
+
+import type {CrDialogElement} from 'chrome://resources/cr_elements/cr_dialog/cr_dialog.js'
+import {WebUiListenerMixin} from 'chrome://resources/cr_elements/web_ui_listener_mixin.js'
+import {assert} from 'chrome://resources/js/assert.js'
+import {PolymerElement} from 'chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js'
+
+import type {SearchEngine} from '../search_engines_page/search_engines_browser_proxy.js'
+import type {BraveSearchEnginesPageBrowserProxy} from './brave_search_engines_page_browser_proxy.js'
+import {BraveSearchEnginesPageBrowserProxyImpl} from './brave_search_engines_page_browser_proxy.js'
+
+import {getTemplate} from './private_search_engine_list_dialog.html.js'
+
+export interface SettingsPrivateSearchEngineListDialogElement {
+  $: {
+    dialog: CrDialogElement,
+  }
+}
+
+const SettingsPrivateSearchEngineListDialogElementBase =
+  WebUiListenerMixin(PolymerElement)
+
+export class SettingsPrivateSearchEngineListDialogElement extends
+    SettingsPrivateSearchEngineListDialogElementBase {
```

Comment:
sorry to do this to you @emerick but could you make this a `LitElement`? I think we should be avoiding adding any new Polymer components (and using Lit elements from inside Polymer is fine).

Reply:
So, I just about converted the entire dialog to a `LitElement`. The problem I ran into at the end (of course), is that this dialog needs the styling provided by `settings_shared.css` and there is no Lit-based equivalent of that at the moment. I'd hate to duplicate all of that code just for this. Is there some other way to get at that CSS? I also noticed that currently it doesn't seem like any of the upstream settings pages use LitElement, so maybe this is part of the reason?

I'm still open to doing this, but just not sure how to get past this particular stumbling block.

Reply:
Yeah, so this should definitely be possible in Lit - you just need to work out what the relative locations of the files are in the generated `preprocessed` directory.

`br_toolbar` has an example of this with `br_shared_style`:

```css
/* #css_wrapper_metadata_start
 * #type=style-lit
 * #import=../br_shared_style.css.js
 * #css_wrapper_metadata_end */ 
 ```

alternatively, if `br_settings_shared` is served from `chrome://resources` somewhere you can add a link[rel] to the HTML source of the WebComponent :smile:

Reply:
Ah yeah, you need an import and an include sorry!

```css
/* #css_wrapper_metadata_start
 * #type=style-lit
 * #import=../br_settings_shared.css.js
 * #include=br-shared-settings
 * #css_wrapper_metadata_end */ 
```

Reply:
I think `ui/webui/resources/cr_elements/cr_button/cr_button.css` (from upstream) has an example

Reply:
Okay I was wrong and we can't quite convert this to Lit yet (the settings page doesn't support Lit in the shared_css). Sorry @emerick!